stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -50,22 +85,70 @@ internal class PaymentSheetViewModel internal constructor(
     }
 
     fun updatePaymentMethods(intent: Intent) {
-        val args: PaymentSheetActivityStarter.Args? = PaymentSheetActivityStarter.Args.fromIntent(intent)
-        if (args == null) {
-            onError(IllegalStateException("Missing activity args"))
-        } else {
+        getPaymentSheetActivityArgs(intent)?.let { args ->
             updatePaymentMethods(
                 args.ephemeralKey,
                 args.customerId
             )
         }
     }
 
+    fun checkout(activity: Activity) {
```

Comment:
Any advantage to using `ComponentActivity` instead of just `Activity`?

Reply:
Can't think of any at the moment. Should be simple to change if we ever need to though

Reply:
Ok, was wondering if we could use `ViewModel` here

Reply:
ah instead of `AndroidViewModel`? I think we can actually